### ISSUE TITLE
Lockwise upload not working 

### ISSUE THREAD

**Issue by nimser** [avallete/lockwise-password-importer#8]
Firefox Version: 76.0.1
OS: Arch Linux

I'm I followed all the steps, sucessfully generated the CSV with `chrome_login_database_extractor.py`, then clicked the email confirmation link and went all the way to the last `INFO: Done !`, `about:logins` but doesn't show the imported entries.

Tried hitting « Sync now » to no avail.

**Comment by avallete** [comment]
Hi there, thanks for your feedback.

I have some questions to help me nail down your issue:

1. Have you double checked that 2FA isn't enabled on your Firefox account?
2. Did you see any error message into the logs?
3. Approximately how much passwords do you try to upload? Does the message (`Synced passwords: X`) just before the `Done!` message match this number?
4. Did you try to logout/login again into your Firefox account?

**Comment by nimser** [comment]
1. Yes, I disabled it.
2. It'm seeing one ERROR and one DEBUG message for the
`chrome_login_database_extractor.py` script, who seem related to each
other. However checking the resulting .csv file my passwords are there so I
don't think they are a problem? Here are the messages:

```
- ERROR: while trying to retrieving decryption key from secretstorage: The
name org.freedesktop.secrets was not provided by any .service files [l.55]
- DEBUG: Cannot retrieve decryption key from secretstorage, use default
linux 'peanuts' key [l.56]
```

3. The scan in `chrome_login_database_extractor.py` finds 449 passwords.
When I run the importer script it says it will import 430 passes because
some of the entries have an empty password field (a message shows: `X
cannot be loaded because length of value must be at least 1 for dictionary
value @ data['password']`
4. I did try that, and redoing the process again, to no avail, even though
the script was running until `Done!`.


I retried this today to answer your message, but something new happened.
Instead of the usual `Done!` at the end of the importer script, I get an
ERROR message. These are the last two lines I get:

```
430 passwords will be loaded into firefox, do you confirm ?  [y/N]: y
ERROR: The request was blocked for security reasons [l.196]
```

**Comment by nimser** [comment]
1. Yes, I disabled it.
2. It'm seeing one ERROR and one DEBUG message for the`chrome_login_database_extractor.py` script, who seem related to each other. However checking the resulting .csv file my passwords are there so I don't think they are a problem? Here are the messages:

```
- ERROR: while trying to retrieving decryption key from secretstorage: The name org.freedesktop.secrets was not provided by any .service files [l.55]
- DEBUG: Cannot retrieve decryption key from secretstorage, use default linux 'peanuts' key [l.56]
```

3. The scan in `chrome_login_database_extractor.py` finds 449 passwords. When I run the importer script it says it will import 430 passes because some of the entries have an empty password field (a message shows: `X cannot be loaded because length of value must be at least 1 for dictionary
value @ data['password']`
4. I did try that, and redoing the process again, to no avail, even though the script was running until `Done!`.


I retried this today to answer your message, but something new happened.
Instead of the usual `Done!` at the end of the importer script, I get an
ERROR message. These are the last two lines I get:

```
430 passwords will be loaded into firefox, do you confirm ?  [y/N]: y
ERROR: The request was blocked for security reasons [l.196]
```

**Comment by avallete** [comment]
Hi,

After further investigation, I've been able to reproduce your bug.

It seem that something has change on the Firefox side, after some debug:

1. The "request blocked" error is throw when we try to `login` to firefox api, now, it seem that I need to get an "unlock code" first (with a mail sending) to retrieve session token. I've already implemented a fix for that.
2. After debug, It seem that the records ARE in fact correctly sent to the firefox api (when running with debug, on my tests, I see the `+500 passwords` when running `get_records('passwords')`. However, for an unknown reason, when syncing, those passwords aren't imported into Firefox.

After some research, I have no clue of what's going on, so I've opened an issue on [bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1640123). I may also try to reach some mozzilla devs via IRC to get more info.

I'm working on a fix here: https://github.com/avallete/lockwise-password-importer/pull/9
